app/models/rule.rb

```diff
@@ -31,6 +31,8 @@ class Rule < ApplicationRecord
     joins(:rule_identifier).where(rule_identifiers: { label: identifier_label })
   }
 
+  default_scope { where(id: RuleResult.pluck(:rule_id)) }
```

Comment:
default_scope?? Why not make a normal scope:

```suggestion
    scope :with_results, lambda { |rule_result_ids = RuleResult.pluck(:id)|
      where(id: rule_result_ids)
    }
```